src/toxcore/dht/client.rs

```diff
@@ -125,12 +127,65 @@ impl Client {
         let nat_ping_req = DhtPacket::DhtRequest(DhtRequest::new(&self.precomputed_key.clone(), &friend_pk, &self.pk, payload));
         self.send(nat_ping_req)
     }
+    // get broadcast addresses for host's network interfaces
+    fn get_ipv4_broadcast_addrs() -> Vec<SocketAddr> {
+        let ifs = pnet_datalink::interfaces();
+        let mut res = Vec::new();
+        ifs.iter().for_each(|interface| 
+            interface.ips
+                .iter()
+                .filter_map(|ip| {
+                    match ip.ip() {
+                        IpAddr::V4(ip) => Some(ip.octets()),
+                        _ => None,
+                    }
+                })
+                .for_each(|oct| {
+                    let ipaddr_str = format!("{}.{}.{}.{}", oct[0], oct[1], oct[2], oct[3]);
+                    let broad = IPAddress::parse(ipaddr_str).unwrap().broadcast().to_string();
+                    let broad_str: Vec<&str> = broad.split('/').collect();
+                    let broad_str= broad_str[0].to_owned() + ":33445";
+                    let sock: SocketAddr = broad_str.parse().unwrap();
+                    res.push(sock);
```

Comment:
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.flat_map

Mutable variable `res` is a bad design.

Reply:
Well, now I focus at CI build pass.